straxen/plugins/peak_processing.py

```diff
@@ -331,98 +336,393 @@ def find_n_competing(peaks, windows, fraction):
+                               num_array, 
+                               sum_array, 
+                               self.config['ambience_divide_t'])
+        result['n_lh_before'] = num_array
+        result['s_lh_before'] = sum_array
+
+        # 4. Calculate number and area sum of small S0, S1, S2 before a peak
+        radius = -1
+        for stype, area in zip([0, 1, 2], 
+                                         self.config['ambience_area_parameters']):
+            num_array = np.zeros(len(current_peak), dtype=np.int16)
+            sum_array = np.zeros(len(current_peak), dtype=np.float32)
+            mask_pre = (peaks['type'] == stype) & (peaks['area'] < area)
+            touching_windows = strax.touching_windows(peaks[mask_pre], roi)
+            # Calculating ambience
+            self.peaks_ambience(current_peak, 
+                                peaks[mask_pre], 
+                                touching_windows, 
+                                radius, 
+                                num_array, 
+                                sum_array, 
+                                self.config['ambience_divide_t'], 
+                                self.config['ambience_divide_r'])
+            result[f'n_s{stype}_before'] = num_array
+            result[f's_s{stype}_before'] = sum_array
+
+        # 5. Calculate number and area sum of small S2 near(in (x,y) space) a S2 peak
+        num_array = np.zeros(len(current_peak), dtype=np.int16)
+        sum_array = np.zeros(len(current_peak), dtype=np.float32)
+        mask_pre = (peaks['type'] == 2) & (peaks['area'] < self.config['ambience_area_parameters'][2])
+        touching_windows = strax.touching_windows(peaks[mask_pre], roi)
+        # Calculating ambience
+        self.peaks_ambience(current_peak, 
+                            peaks[mask_pre], 
+                            touching_windows, 
+                            self.config['ambient_radius'], num_array, 
+                            sum_array, 
+                            self.config['ambience_divide_t'], 
+                            self.config['ambience_divide_r'])
+        result['n_s2_near'] = num_array
+        result['s_s2_near'] = sum_array
+
+        # 6. Set time and endtime for peaks
+        result['time'] = current_peak['time']
+        result['endtime'] = strax.endtime(current_peak)
+        return result
+
+    @staticmethod
+    @numba.njit
+    def lonehits_ambience(peaks, pre_hits, 
+                          touching_windows, num_array, 
+                          sum_array, ambience_divide_t):
+        # Function to find lonehits before a peak
+        # creating_hit is the lonehit creating ambience
+        # suspicious_peak is the suspicious peak in the ambience created by creating_hit
+        for p_i, suspicious_peak in enumerate(peaks):
+            indices = touching_windows[p_i]
+            for idx in range(indices[0], indices[1]):
+                creating_hit = pre_hits[idx]
```

Comment:
According to @GiovanniVolta's study, sometimes there are lone_hits with negative area (See [this plot ](https://xe1t-wiki.lngs.infn.it/lib/exe/fetch.php?cache=&media=xenon:giovo:led_vs_lh_spe_area.png) or [this note](https://xe1t-wiki.lngs.infn.it/doku.php?id=xenon:giovo:led_calibration_vs_lone_hits)), so probably it might be better to just use lone_hits which have area > 0